### PR TITLE
[NUI] Replace XXXList with IList<XXX>.

### PR DESCRIPTION
**Pull request by huayongxu**
### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs [hunk 1]
```diff
@@ -73,54 +73,45 @@ public string OriginalUrl
...
         /// <summary>
-        /// Get item count.
+        /// Dispose for IDisposable pattern
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public uint ItemCount
+        protected override void Dispose(DisposeTypes type)
         {
-            get
+            if (disposed)
```
[the remark this answers is not in the document]
Comment (huayongxu) [reply src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs]: Done

File: src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs [hunk 2]
```diff
@@ -180,35 +168,92 @@ public WebBackForwardListItem GetItemAtIndex(int index)
...
+                list.Add(backwardItemList.GetItemAtIndex(i));
+            }
+            return list;
         }
 
         /// <summary>
         /// Get copied forward items.
         /// </summary>
-        /// <param name="index">limit The number of items to retrieve.</param>
+        /// <param name="limit">The number of items to retrieve.</param>
```
[the remark this answers is not in the document]
Comment (huayongxu) [reply src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs]: Done.

File: src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs [hunk 3]
```diff
@@ -180,35 +168,92 @@ public WebBackForwardListItem GetItemAtIndex(int index)
...
-            WebBackForwardListItem ret = new WebBackForwardListItem(itemPtr, true);
-            return ret;
+            return new WebBackForwardListItem(itemPtr, true);
         }
 
         /// <summary>
         /// Get copied backward items.
         /// </summary>
-        /// <param name="index">limit The number of items to retrieve.</param>
+        /// <param name="limit">The number of items to retrieve.</param>
```
[the remark this answers is not in the document]
Comment (huayongxu) [reply src/Tizen.NUI/src/internal/WebView/WebBackForwardList.cs]: Done.

File: src/Tizen.NUI/src/internal/WebView/WebContext.cs [hunk 4]
```diff
@@ -42,12 +45,41 @@ internal WebContext(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cM
             passwordDataListAcquiredProxyCallback = OnPasswordDataListAcquired;
         }
 
+        /// <summary>
+        /// Dispose for IDisposable pattern
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void Dispose(DisposeTypes type)
+        {
+            if (disposed)
```
[the remark this answers is not in the document]
Comment (huayongxu) [reply src/Tizen.NUI/src/internal/WebView/WebContext.cs]: Done

**Review by AchoWang — APPROVED** [review]
Looks good to me.

**Review by zhouleonlei — APPROVED** [review]
LGTM

**Review by elishateng — APPROVED** [review]
LGTM

**Review by huiyueun — APPROVED** [review]

**Review by zg2nets — APPROVED** [review]
looks good to me！

**Review by rabbitfor — APPROVED** [review]
LGTM

**Pull request merged**